test/socket.js

```diff
@@ -311,3 +312,284 @@ test('try send after close', async function (t) {
+
+  const u = new UDX()
+  const a = u.createSocket()
+
+  a.bind(0)
+
+  const invalidHost = '0.-1.0.0'
+
+  try {
+    a.trySend(Buffer.from('hello'), a.address().port, invalidHost)
+  } catch (error) {
+    t.is(error.message, `${invalidHost} is not a valid IP address`)
+  }
+
+  await a.close()
+})
+
+test('send without bind', async function (t) {
+  t.plan(1)
+
+  const u = new UDX()
+
+  const a = u.createSocket()
+  const b = u.createSocket()
+
+  b.on('message', function (message) {
+    console.log('recv')
+    t.alike(message, Buffer.from('hello'))
+    a.close()
+    b.close()
+  })
+
+  b.bind(0)
+  await a.send(Buffer.from('hello'), b.address().port)
+})
+
+test('try send without bind', async function (t) {
+  t.plan(1)
+
+  const u = new UDX()
+
+  const a = u.createSocket()
+  const b = u.createSocket()
+
+  b.on('message', function (message) {
+    console.log('recv')
+    t.alike(message, Buffer.from('hello'))
+    a.close()
+    b.close()
+  })
+
+  b.bind(0)
+  a.trySend(Buffer.from('hello'), b.address().port)
+})
+
+test('get address without bind', async function (t) {
+  const u = new UDX()
+  const a = u.createSocket()
+  t.is(a.address(), null)
+})
+
+test('bind twice', async function (t) {
+  t.plan(1)
+
+  const u = new UDX()
+  const a = u.createSocket()
+
+  a.bind(0)
+
+  try {
+    a.bind(0)
+  } catch (error) {
+    t.is(error.message, 'Already bound')
+  }
+
+  await a.close()
+})
+
+test('bind while closing', async function (t) {
+  t.plan(1)
+
+  const u = new UDX()
+  const a = u.createSocket()
+
+  a.close()
+
+  try {
+    a.bind(0)
+  } catch (error) {
+    t.is(error.message, 'Socket is closed')
+  }
+})
+
+test('close twice', async function (t) {
+  t.plan(1)
+
+  const u = new UDX()
+  const a = u.createSocket()
+
+  a.bind(0)
+
+  const closing1 = a.close()
+  const closing2 = a.close()
+
+  t.ok(closing1 === closing2)
+})
+
+test('set TTL', async function (t) {
+  t.plan(2)
+
+  const u = new UDX()
+  const a = u.createSocket()
+
+  a.on('message', function (message) {
+    t.alike(message, Buffer.from('hello'))
+    a.close()
+  })
+
+  try {
+    a.setTTL(5)
+  } catch (error) {
+    t.is(error.message, 'Socket not active')
+  }
+
+  a.bind(0)
+  a.setTTL(5)
+
+  await a.send(Buffer.from('hello'), a.address().port)
+})
+
+test('get recv buffer size', async function (t) {
+  t.plan(2)
+
+  const u = new UDX()
+  const a = u.createSocket()
+
+  try {
+    a.getRecvBufferSize()
+  } catch (error) {
+    t.is(error.message, 'Socket not active')
+  }
+
+  a.bind(0)
+  t.ok(a.getRecvBufferSize() > 0)
+
+  await a.close()
+})
+
+test('set recv buffer size', async function (t) {
+  t.plan(2)
+
+  const u = new UDX()
+  const a = u.createSocket()
+
+  const NEW_BUFFER_SIZE = 8192
+
+  try {
+    a.setRecvBufferSize(NEW_BUFFER_SIZE)
+  } catch (error) {
+    t.is(error.message, 'Socket not active')
+  }
+
+  a.bind(0)
+  a.setRecvBufferSize(NEW_BUFFER_SIZE)
+
+  t.is(a.getRecvBufferSize(), NEW_BUFFER_SIZE)
```

Comment:
Api does not guarantee this, due to libuv not guaranteeing this. Test that it’s larger or equal or remove this test